src/libraries/Microsoft.Extensions.Logging.Configuration/src/LoggingBuilderConfigurationExtensions.cs

```diff
@@ -14,6 +15,9 @@ public static class LoggingBuilderConfigurationExtensions
         /// Adds services required to consume <see cref="ILoggerProviderConfigurationFactory"/> or <see cref="ILoggerProviderConfiguration{T}"/>
         /// </summary>
         /// <param name="builder">The <see cref="ILoggingBuilder"/> to register services on.</param>
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2026:RequiresUnreferencedCode",
+            Justification = "ServiceDescriptor has RequiresUnreferencedCode because Service and Implementation types could be generic with mis-matching trimming annotations. " +
```

Comment:
```suggestion
            Justification = "TryAddSingleton has RequiresUnreferencedCode because Service and Implementation types could be generic with mis-matching trimming annotations. " +
```